shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -606,7 +605,14 @@ class JSTestBackend extends JSBackend {
       case term: Term =>
         try {
           val body = translateTerm(term)(scope)
-          JSTestBackend.CodeQuery(scope.tempVars.emit(), (resultIdent := body) :: Nil)
+          val code = body match {
+            case JSIdent(name) if (name.indexOf('\'') > -1) => (resultIdent := JSIdent(s"globalThis[\"${name}\"]"))
```

Comment:
This looks like a _huge_ hack. Making this logic rely on the fresh name encoding of the `Scope` class is totally brittle (what if we want to change it later?). You should add the correct logic earlier, nearer to the place where this thing is decided.

Reply:
Wait so this logic was not needed at all?

Reply:
I found something wrong in `shared\src\test\diff\mlscript\BadInherit2.mls`, where:
```
f'
//│ res: 1
//│    = A1 { Foo1: [Function (anonymous)] }
```

Initially, I thought it might be a mistake because `f'` should be `1`, so I added this check to avoid it. However, I found it wrong in the original file:
```
f'
//│ res: 1
//│    = 0
```

So I think maybe it should not print `1` at all, and this check is redundant (or at least should be fixed in another pr).

Reply:
Ah I see! This is actually an instance of https://github.com/hkust-taco/mlscript/issues/65#issuecomment-1168257197

It works without the tick:

```scala
def f_ = 1
//│ f_: 1
//│   = 1

f_
//│ res: 1
//│    = 1
```

We need to replace ticks in JS with some other symbol. This should be doable with a small modification to the scope class. Could you do it in this PR – or in another one if you prefer?

Reply:
I can fix it in another PR later.  😄

Reply:
Ok cool, please do!